### PR TITLE
add Crystal

### PR DESCRIPTION
**Pull request by ubnt-intrepid**
[Crystal](https://crystal-lang.org/) is statically typed, Ruby-like programming language.

Build scripts are based on:
* https://crystal-lang.org/docs/installation/on_debian_and_ubuntu.html
* https://crystal-lang.org/docs/installation/from_source_repository.html
* https://github.com/crystal-lang/crystal/wiki/All-required-libraries#ubuntu

### Notes
* Depenencies for `libevent-dev`, `libgc-dev` and `libpcre3-dev` are required at compilation (see also `test/test-server/Dockerfile`)
  - According to [here](https://github.com/crystal-lang/crystal/wiki/All-required-libraries#ubuntu), more packages may be required...

### CODE REVIEW

File: build/crystal/VERSIONS [hunk 1]
```diff
@@ -0,0 +1,2 @@
+0.20.5
+0.21.1
```
Comment (melpon) [inline build/crystal/VERSIONS]: Sort by newest.

**Comment by melpon** [comment]
Add execution permission:

- `crystal/install.sh`
- `crystal/test.sh`
- `crystal-head/install.sh`
- `crystal-head/testl.sh`

**Comment by ubnt-intrepid** [comment]
@melpon OK, done.

**Comment by melpon** [comment]
Merged. Thank you for your contribution!
Please send a pull request to https://github.com/wraith13/wandbox-vscode like Pony.

**Comment by ubnt-intrepid** [comment]
All right.

**Pull request merged**